allennlp/models/simple_tagger.py

```diff
@@ -0,0 +1,137 @@
+        A Vocabulary, required in order to compute sizes for input/output projections.
+    embedding_dim : int, optional (default = 100)
+        The dimensionality of the embedding space used to embed the input sequence.
+    hidden_size : int, optional (default = 200)
+        The dimensionality of the hidden state of the LSTM encoder.
+    num_layers : int, optional (default = 2)
+        The number of stacked LSTM encoders to use.
+    """
+
+    def __init__(self,
+                 vocabulary: Vocabulary,
+                 embedding_dim: int = 100,
+                 hidden_size: int = 200,
+                 num_layers: int = 2):
+        super(SimpleTagger, self).__init__()
+
+        self.vocabulary = vocabulary
+        self.embedding_dim = embedding_dim
+        self.hidden_size = hidden_size
+        self.num_layers = num_layers
+        self.num_classes = self.vocabulary.get_vocab_size("tags")
+
+        self.embedding = Embedding(self.vocabulary.get_vocab_size("tokens"),
+                                   self.embedding_dim)
+        # TODO(Mark): support masking once utility functions are merged.
+        self.stacked_encoders = LSTM(self.embedding_dim,
+                                     self.hidden_size,
+                                     self.num_layers,
+                                     batch_first=True)
+        self.tag_projection_layer = TimeDistributed(Linear(self.hidden_size,
+                                                           self.num_classes))
+        self.sequence_loss = torch.nn.CrossEntropyLoss()
+
+    def forward(self,  # pylint: disable=arguments-differ
+                sequence_tokens: torch.IntTensor,
+                sequence_tags: torch.IntTensor = None):
+        """
+        Parameters
+        ----------
+        sequence_tokens : torch.LongTensor, required
+            A torch tensor representing the indices of the
+        sequence_tags : torch.LongTensor, optional (default = None)
+            A torch tensor representing the sequence of gold labels.
+            These can either be integer indexes or one hot arrays of
+            labels, so of shape (batch_size, sequence_length) or of
+            shape (batch_size, sequence_length, vocabulary_size).
+
+        Returns
+        -------
+        An output dictionary consisting of:
+        logits : torch.FloatTensor
+            A tensor of shape (batch_size, sequence_length, tag_vocab_size)
+            representing unnormalised log probabilities of the tag classes.
+        loss: : torch.FloatTensor, optional
+            A scalar loss to be optimised.
+
+        """
+        batch_size = sequence_tokens.size()[0]
+        # TODO(Mark): Change to use NlpApi.
+        embedded_text_input = self.embedding(sequence_tokens)
+        encoded_text, _ = self.stacked_encoders(embedded_text_input)
+
+        logits = self.tag_projection_layer(encoded_text)
+        reshaped_log_probs = logits.view(-1, self.num_classes)
```

Comment:
No - also we don't want to do that here, because we the cross entropy loss function takes unnormalised log probabilities for stability reasons.